cli/tests/integration_tests.rs

```diff
@@ -79,10 +80,16 @@ fn mine_sapling_to_self() {
     let (_regtest_manager, _child_process_handler, client, runtime) =
         setup::coinbasebacked_spendcapable();
 
-    runtime.block_on(client.do_sync(true)).unwrap();
+    runtime.block_on(async {
+        sleep(Duration::from_secs(2)).await;
+        client.do_sync(true).await.unwrap();
 
-    let balance = runtime.block_on(client.do_balance());
-    assert_eq!(balance["sapling_balance"], 625000000);
+        let balance = client.do_balance().await;
+        println!("{}", json::stringify_pretty(balance.clone(), 4));
```

Comment:
If we're bothering to check state (by printing) then..   I think that either gets removed (it was just a debug for dev) or becomes some sort of assertion (this should be true in this scenario generally) when the test is "mature".

I generally prefer that latter.   I like for the data that came from the print (or some small meaningful part of it) to be checked in.  That makes the nature of the test easy to read.   Of course...  if it's a ton of data, or if it varies from run-to-run then it might not be beneficial to shove it in.   Too hard to read!

Reply:
It was a debug for dev that got missed. Currently I have a minimal set of asserts to check what the test is intending on checking, it would probably be smart to toss in some more assertions, up to everything that we know should be true.